.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [14]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup node env
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run type-check and ESLint
+        run: npx lint-staged
```

Comment:
did you test if this returns non-zero status code if there are linting errors?

Reply:
yes I did

Reply:
why does this step say `No staged files found.`, does `lint-staged` work in the CI since we're not staging anything?

Reply:
It says that because there were no TS/JS files changed. I'm not sure what you mean by the second question.

Reply:
From what I understand, `lint-staged` seems to be used before you commit changes, when the changes are staged by git. When you're running CI, is anything staged? I'm not sure if this library is meant to be used in the CI. But if you say it works for catching errors then sure

Reply:
I'd be more comfortable if you push a commit with linting errors and seeing the github action fail on that commit